### DISCUSSION TITLE
MudBlazor

### DISCUSSION THREAD

**Discussion by medbour** [oqtane/oqtane.framework#1819] [Q&A] [answered]
1-is it possible to use MudBlazor library in a module 
2-what will be the best method
   install the mudblazor in each module or install it in the framwork level

**Comment by leigh-pointer** [comment, marked as answer]
@medbour 
The best thing would be to create a theme with MudBlazor references because MudBlazor is Bootstrap extension.  

I have played with trying to overload the current page theme but results where not good.  (I could be wrong)

**Reply by harperjohn** [reply]
Any more info or progress on using MudBlazor?

**Reply by zhengdayongbob** [reply]
I added MudBlazor into Oqtane:

Step1: add the below blazor css, js file in Oqtane.Server\wwwroot\Themes\Oqtane.Themes.MudBlazorTheme\

![image](https://user-images.githubusercontent.com/11614817/152405974-0243dd14-765c-4bce-ad67-3b8a824a9c21.png)

Step 2: At Client side, add \Oqtane.Client\Themes\MudBlazorTheme\ all files as below:
![image](https://user-images.githubusercontent.com/11614817/152406542-908e8d0c-142a-4d90-a104-bbf172a6ffde.png)

Step 3: at  Oqtane.Server\Startup.cs, add the below code:

            using MudBlazor;
            using MudBlazor.Dialog;
            using MudBlazor.Services;
            at function: public void ConfigureServices(IServiceCollection services)
            // Register Services for Mudblazor Theme
            services.AddMudBlazorDialog();
            services.AddMudBlazorSnackbar();
            services.AddMudBlazorResizeListener();

Step 4: in page, set a page use the MudBlazor theme.

**Comment by Loclhh** [comment]
> 1-is it possible to use MudBlazor library in a module 2-what will be the best method install the mudblazor in each module or install it in the framwork level

Hi Medbour, did you success apply mudblazor in oqtane projects? If you are success, could you sumarize your steps for other people can follow?
Thank in advance!

**Reply by zhengdayongbob** [reply]
see my above replay. I will try to have a demo code based on 3.0 version.

**Reply by oqtanecms** [reply]
> see my above replay. I will try to have a demo code based on 3.0 version.

Could you share me your demo code based on latest Oqtane version?

**Comment by sbwalker** [comment]
@Loclhh it should be possible to install MudBlazor as a standard external Oqtane module ie. without modifying any files in the core framework. Basically, you need to create a module, include the MudBlazor DLLs and static assets, and implement the IServerStartup interface to register the startup services.

**Reply by ricardok1** [reply]
Hi @sbwalker 
Can you please detail step by step on this?
Thanks
